### ISSUE TITLE
[Tenserflow]_[Loss function]

### ISSUE THREAD

**Issue by jangsik-park** [jangsik-park/Study#9]
# [1] Loss function이란 무엇일까요? (n412)

- 손실함수는 현재의 가중치 세트를 평가하는데 사용하는 함수 입니다.

- 예측값과 실제값의 차이를 표현하는 지표 입니다.

- 즉, 모델 성능의 '나쁨' 을 나타내는 지표로, '모델이 데이터를 얼만큼 잘 처리하지 못 했는가?' 를 나타내는 지표 입니다.

# [2] Loss fuction의 종류

0. 크게 회귀 타입과 분류 타입의 손실함수로 나뉘어 집니다.
회귀타입에는 [MAE], [MSE], [RMSE]
분류 타입에는 [Binary_Crossentropy], [Categorical Crossentropy], [Sparse Categorical Crossentropy]  가 있습니다.

ㄱ. binary_Crossentropy

- 이중분류에 사용합니다.
- 예측값이 0과 1사이의 확률 값으로 나옵니다.
- keras에서는 model.predict_classes(x).flatten ( flatten 생략 가능) 로 각 데이터에 대한 class를 구할 수 있습니다.
-  ex) loss='binary_crossentropy'

ㄴ. Categorical Crossentropy

- 다중분류에 사용합니다.
- 출력값이 Onehot-Encoding된 형태로 나옵니다. 따라서 라벨=y를  one-hot 형태로 제공해줘야 합니다.
- 클래스가 상호 배타적일 경우(e.g. 각 샘플이 정확히 하나의 클래스에 속하는 경우) 사용 합니다.
- ex) loss='categorical_crossentropy'

ㄷ. Sparse Categorical Crossentropy

- 다중분류에 사용합니다.
- 샘플 값이 정수형 자료입니다.
- 즉, 샘플 값을 입력하는 부분에서 **별도의 Onehot-Encoding을 하지 않고** 정수값 그대로 줄 수 있습니다.
- 한 샘플에 여러 클래스가 있거나 label이 soft 확률일 경우 사용 합니다.
-  ex) loss='sparse_categorical_crossentropy'

ㄹ. MeanSquaredError

- 예측한 값과 실제 값 사이의 평균 제곱 오차를 정의한다. 
- 차가 커질수록 제곱 연산으로 인해서 값이 더욱 뚜렷해진다.
-  제곱으로 인해서 오차가 양수이든 음수이든 누적 값을 증가시킨다.
-  ex) loss='mean_squared_error'

[참고자료1](https://durian9s-coding-tree.tistory.com/2)
[참고자료2](https://hororolol.tistory.com/375)
